valhalla/mjolnir/landmark_database_builder.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <boost/property_tree/ptree.hpp>
+#include <cstdint>
+#include <iostream>
+#include <list>
+#include <string>
+#include <vector>
+
+#include "baldr/graphconstants.h"
+#include "filesystem.h"
+#include "mjolnir/util.h"
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include <geos_c.h>
+
+namespace valhalla {
+namespace mjolnir {
+enum class AccessMode { ReadWriteCreate, ReadOnly, ReadWrite };
```

Comment:
this isnt needed. the constructor should take a single boolean that maybe like: `read_only = true`

then in the constructor we can figure out the right thing to do automatically:

```pseudo code
if !exists(file)
  if read_only
    throw invalid option
  open_db(SQLITE_CREATE_FLAG | SQLITE_READWRITE_FLAG)
else
  open_db(read_only ? SQLITE_READONLY_FLAG : SQLITE_READWRITE_FLAG)
```

Reply:
I tried to use `std::ifstream` to decide if file exists but it didn't work very well. Do you have a better suggestion?